lib/statsd/instrument/prometheus/aggregators/timing.rb

```diff
@@ -4,15 +4,108 @@ module StatsD
   module Instrument
     module Prometheus
       module Aggregators
+        # Borrowed from: https://github.com/statsd/statsd/blob/master/lib/process_metrics.js#L22
+        # Read more about the format here: https://github.com/statsd/statsd/blob/master/docs/metric_types.md#timing
         class Timing < Base
           def aggregate
-            # TODO
-            datagrams.last
+            current_timer_data = { }
+            current_timer_count_data = { }
+            values = datagrams.map(&:value).sort
+            count = values.length
+            min = values.first
+            max = values.last
+
+            cumulative_values = [min]
+            cumulative_sum_squares_values = [min * min]
+            for i in 1..(count - 1) do
+              cumulative_values.push(values[i] + cumulative_values[i-1])
+              cumulative_sum_squares_values.push((values[i] * values[i]) + cumulative_sum_squares_values[i - 1])
+            end
+
+            sum = min
+            sum_squares = min * min
+            mean = min
+            threshold_boundary = max
+
+            percentiles.each do |percentile_threshold|
+              count_within_percentile_threshold = count.to_f
+
+              if (count > 1)
+                count_within_percentile_threshold = (percentile_threshold.abs / 100.0 * count).round
+                next if count_within_percentile_threshold == 0
+
+                if (percentile_threshold > 0)
+                  threshold_boundary = values[count_within_percentile_threshold - 1]
+                  sum = cumulative_values[count_within_percentile_threshold - 1]
```

Comment:
similarly, `.take` is generally more ruby-ish, but maybe we want to stay more similar to the original.

Reply:
I agree with both, but I'd like to stay close to the original implementation in case we ever want to port over any updates. It will be easier to port updates if the source is closer.